### PR TITLE
Publish Storybook on Chromatic

### LINKED ISSUE #58: Publish Storybook as docs / for review, examples
We should make the spark Storybook available not only in local dev, but also statically host it so at least any team member can view it. ([Storybook's general guidance on this](https://storybook.js.org/docs/react/workflows/publish-storybook))

### Options

[storybook-deployer](https://github.com/storybookjs/storybook-deployer)
- completely free, open-source tool
- can use an existing build (gen'd by nx)
- can push to GitHub Pages or AWS
- [Setup Guidance](https://github.com/storybookjs/storybook-deployer#storybook-deployer)

[Chromatic](https://www.chromatic.com/)
- generous free tier
  - [pricing differences](https://www.chromatic.com/pricing#FeaturesSection) are for "snapshots", i.e. catching visual regressions
- integrates with GitHub to run on PR's, detect changes, and start a "Visual" Review process
  - review: before/after comparison, discussion thread, approval mechanism
  - syncs repo "Collaborators" to participate in this review -- and for plain viewing as well
- [Setup guidance](https://www.chromatic.com/docs/storybook)

### Pros v Cons

storybook-deployer
- pro: can deploy to GitHub pages
  - very natural integration for repo visitors, and keeps our relevant domains to 1.
- con: Abandoned repo
  - no meaningful activity for about a year -- it _was_ created and maintained by the core storybook team, but several flags straight up don't work, no maintainers respond, a bot just upgrades pkgs
  - core function works fine however, see an [initial deployment](https://prenda-school.github.io/prenda-spark/?path=/story/prenda-spark-iconbutton--configurable-input)

Chromatic
- pro: [viewable build history](https://www.chromatic.com/builds?appId=60b7f55f55fbd4004993da4c)
- pro: [integrates with GitHub PRs](https://www.chromatic.com/pullrequests?appId=60b7f55f55fbd4004993da4c)
  - setup CI/CD pipeline
  - "[UI Review](https://www.chromatic.com/docs/review)" feature scans PRs for new / changes ([UI Tests](https://www.chromatic.com/docs/test)) (for multiple browsers too!) and injects "Review" into checks for PR's to pass.
- con: the review feature might stop when we hit the free tier $5k snapshot limit?

### Concluding Thoughts

The integrated PR UI Review is the killer feature of Chromatic, I don't think it gets much better than that.

For either option, the ability to build a storybook for multiple projects in a monorepo is absent. I think this means we should think about extracting our storybook into a separate project.

### PR DESCRIPTION
**Pull request by WilliamKelley**
Closes #58 

Tasks
- [x] have Chromatic's access request to org approved
- [x] add script & do initial deployment
- [x] fix fonts not being in build
- [x] have Chromatic's access request to spark PR's approved
- [x] add CI step with token as a secret & remove script

### CODE REVIEW

**Comment by WilliamKelley** [comment]
The deploy to chromatic check is failing. I'm communicating with support to figure out why the generated project url / token is connected correctly to our account -- as indicated in the runner console error. Hopefully they can resolve it this week.

File: .github/workflows/chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: 'Chromatic'
+
+on: [push, workflow_dispatch]
+
```
Comment (shadoath) [inline .github/workflows/chromatic.yml:4]: Do we want this limited to the `main` branch too?

Comment (WilliamKelley) [reply .github/workflows/chromatic.yml:4]: No, this allows it to run on PR's for visual change approvals

Comment (WilliamKelley) [reply .github/workflows/chromatic.yml:4]: see https://www.chromatic.com/docs/github-actions#recommended-configuration-for-build-events

**Review by shadoath — APPROVED** [review]

**Pull request merged**